File: packages/ts-sdk/test/integration/live-api.test.ts
import { describe, expect, it } from 'vitest';
import { TreeDxClient } from '../../src/treedx/index.js';

describe('live TreeDX API integration', () => {
  it('runs health when TREEDX_BASE_URL is configured or reports not configured cleanly', async () => {
    const baseUrl = process.env.TREEDX_BASE_URL;
    if (!baseUrl) {
      expect({ status: 'not_configured', reason: 'TREEDX_BASE_URL is not set' }).toMatchObject({ status: 'not_configured' });
      return;
    }

    const client = new TreeDxClient({ baseUrl, token: process.env.TREEDX_TOKEN });
    await expect(client.health()).resolves.toBeDefined();
  });
});

File: packages/elixir-sdk/test/integration/live_api_test.exs
defmodule TreeDxSdk.LiveApiTest do
  use ExUnit.Case, async: true

  test "live health is optional" do
    case System.get_env("TREEDX_BASE_URL") do
      nil ->
        assert true

      base_url ->
        client = TreeDxSdk.Client.new(base_url: base_url, token: System.get_env("TREEDX_TOKEN"))
        assert {:ok, _} = TreeDxSdk.health(client)
    end
  end
end
